### ISSUE TITLE
[jacobsen on 2010-02-02 17:30:42] : Logging setup failure, which kills runs

### ISSUE THREAD

**Issue by dglo** [dglo/pdaq#1270]
See note

Additional Information: On Jan 31, 2010, at 10:09 PM, Koskinen, David (Guest) wrote:
The geometry run on DeepCore was unsuccessful due to consistent  
instability in running the pdaq on sps-letrigger, and the detector  
was returned to IC59 data taking. The DAQ on letrigger never  
attained a stable run and cycled through the following restarting  
process ~17 times

Starting run 24...
2010-01-31 08:35:02.768268: STOPPED -> STARTING
2010-01-31 08:35:29.116008: STARTING -> ERROR
2010-01-31 08:35:29.118988: ERROR -> RECOVERING
2010-01-31 08:35:38.157751: RECOVERING -> STOPPED
Starting run 25...
2010-01-31 08:35:38.162176: STOPPED -> STARTING
2010-01-31 08:35:51.189118: STARTING -> ERROR
2010-01-31 08:35:51.192780: ERROR -> RECOVERING
2010-01-31 08:36:00.260127: RECOVERING -> STOPPED
Starting run 26...
2010-01-31 08:36:00.264499: STOPPED -> STARTING
2010-01-31 08:36:13.292844: STARTING -> ERROR
2010-01-31 08:36:13.296747: ERROR -> RECOVERING
2010-01-31 08:36:22.347732: RECOVERING -> STOPPED
Starting run 27...
2010-01-31 08:36:22.352103: STOPPED -> STARTING
2010-01-31 08:36:35.378654: STARTING -> ERROR
2010-01-31 08:36:35.382314: ERROR -> RECOVERING
2010-01-31 08:36:44.440097: RECOVERING -> STOPPED

Lisa had problems this morning with letrigger where the parallel DAQ  
launch took 7 cycles before reaching a stable state, which deviates  
from our past experience with letrigger in that it hasn't ever had  
issues with at least starting a stable run. Hopefully it will be in  
a better mood tomorrow when further attempts are made at Flashing.

Dave G reports:

These runs all have really strange errors:

  Catchall - [2010-01-31 08:36:56.608510] Created Run Set #1
  Catchall - [2010-01-31 08:36:56.774922] Failed to start run:  
IOError("[Errno 2] No such file or directory: '/mnt/data/pdaq/log/ 
daqrun00028/dash.log'") in __init__() (DAQLogClient.py:164) <-  
createFileAppender() (DAQRun.py:806) <- setup_run_logging() (DAQRun.py: 
860) <- run_thread() (DAQRun.py:1284)
  Catchall - [2010-01-31 08:36:57.524912] Recovering from failed run  
28...
  Catchall - [2010-01-31 08:36:57.525233] Run terminated WITH ERROR.

For some reason, DAQRun is failing to create the initial pDAQ logging  
directory.  The code is trivial stuff which has been working  
flawlessly.  I'm not sure why it or how it's broken now!
_______________________________________________
logbook mailing list
logbook@icecube.wisc.edu
http://lists.icecube.wisc.edu/mailman/listinfo/logbook

**Comment by dglo** [comment]
[jacobsen on 2010-02-02 17:36:42]
What we found today (Sabrina and I) is that changing the run number to a run which had not been run before (based on the directories in /mnt/data/pdaq/log) seemed to avoid the problem (at least the first and only time we've tried so far).

I suspect there is something slightly goofy with the 'move-existing-log-directories-away' logic in DAQRun.py.  That stuff has been refactored since I wrote the original version.

Since we changed the run number once with ExpControlSkel, it should 'remember' the new sequence and avoid that problem for future runs... if, in fact, it actually WAS the problem.

Here is the command:

./ExpControlSkel.py -c sps-partition-parallel-strings-8-15-23-v2 -n1 -d 1200 -s198\n\n
